solutions/usaco-813.mdx

```diff
@@ -5,54 +5,248 @@ title: Snow Boots
 author: Benjamin Qi
 ---
 
-[Official Analysis](http://www.usaco.org/current/data/sol_snowboots_gold_feb18.html)
+[Official Analysis (C++)](http://www.usaco.org/current/data/sol_snowboots_gold_feb18.html)
 
 We can use a sorted set in place of the linked list mentioned in the official
-solution. The time complexity remains the same.
+solution.
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(N \log N + B \log B)$
+
+<LanguageSection>
+<CPPSection>
+
+We can use a sorted set in place of the linked list mentioned in the official
+solution.
 
 ```cpp
 #include <bits/stdc++.h>
 using namespace std;
 
 int main() {
-	freopen("snowboots.in","r",stdin);
-	freopen("snowboots.out","w",stdout);
+	freopen("snowboots.in", "r", stdin);
+
+	int tile_num;
+	int boot_num;
+	cin >> tile_num >> boot_num;
 
-	int N,B; cin >> N >> B;
-	vector<int> f(N); for (int& t: f) cin >> t;
+	vector<int> tiles(tile_num);
+	for (int& t: tiles) {
+		cin >> t;
+	}
+
+	struct Boot { int max_depth, max_steps, index; };
```

Comment:
why don't you make the struct global?

Reply:
honestly, probably should for consistency